### PR TITLE
Add compile current target action

### PR DESCRIPTION
**Pull request by wisechengyi**
This change adds a quick way to trigger Pants compile for the target under edit. User can choose to tie the action in their keymap.

### CODE REVIEW

File: tests/com/twitter/intellij/pants/integration/OSSPantsCompileActionsTest.java [hunk 1]
```diff
@@ -77,6 +86,36 @@ public void testCompileAllTargetsInModuleAction() throws Throwable {
...
+      String target = classAndTarget.getSecond();
+
+      PsiClass clazz = JavaPsiFacade.getInstance(myProject)
+        .findClass(clazzName, GlobalSearchScope.projectScope(myProject));
+
+      assertNotNull(String.format("%s does not exist, but should.", clazz), clazz);
+
+      FileEditorManager.getInstance(myProject).openFile(clazz.getContainingFile().getVirtualFile(), true);
+      PantsCompileCurrentTargetAction compileCurrentTargetAction = new PantsCompileCurrentTargetAction();
+      Set<String> currentTargets = compileCurrentTargetAction.getTargets(getPantsActionEvent(), myProject).collect(Collectors.toSet());
```
Comment (baroquebobcat) [inline tests/com/twitter/intellij/pants/integration/OSSPantsCompileActionsTest.java]: Could you adjust the whitespace here a bit to make it clearer that the `getTargets` call is what is exercising the code being tested?

File: src/com/twitter/intellij/pants/compiler/actions/PantsCompileCurrentTargetAction.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  protected PantsCompileCurrentTargetAction(String name) {
+    super(name);
+  }
+
+  public PantsCompileCurrentTargetAction() {
+    this("Compile target(s) in the selected editor");
+  }
+
+  /**
+   * Find the target(s) that are only associated with the file opened in the selected editor.
```
Comment (baroquebobcat) [inline src/com/twitter/intellij/pants/compiler/actions/PantsCompileCurrentTargetAction.java]: Based on my reading of the javadoc for FileEditorManager, this will currently run compile for all of the files selected in editor panes. Maybe it should only compile the currently focused one?

File: tests/com/twitter/intellij/pants/integration/OSSPantsCompileActionsTest.java [hunk 3]
```diff
@@ -77,6 +86,36 @@ public void testCompileAllTargetsInModuleAction() throws Throwable {
     assertFalse(compileAllTargetsInModuleAction.doCleanAll());
   }
 
+  public void testCompileTargetsInSelectedEditor() throws Throwable {
+    doImport("examples/tests/scala/org/pantsbuild/example");
```
Comment (baroquebobcat) [inline tests/com/twitter/intellij/pants/integration/OSSPantsCompileActionsTest.java]: Could you add another test that covers what happens if there are multiple editor panes open?

File: src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java [hunk 4]
```diff
@@ -199,6 +199,7 @@ public boolean executeTask(
     // If project has not changed since last Compile, return immediately.
     if (!FileChangeTracker.shouldRecompileThenReset(currentProject, targetAddressesToCompile)) {
       PantsExternalMetricsListenerManager.getInstance().logIsPantsNoopCompile(true);
+      showPantsMakeTaskMessage("Already up to date.\n", ConsoleViewContentType.SYSTEM_OUTPUT, currentProject);
```
Comment (baroquebobcat) [inline src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java]: Could you move this to another review? It seems unrelated.

File: tests/com/twitter/intellij/pants/integration/OSSPantsCompileActionsTest.java (hunk 3 above)
Comment (wisechengyi) [inline tests/com/twitter/intellij/pants/integration/OSSPantsCompileActionsTest.java]: there are two tests classes and their associated targets, so when the 2nd one is opened, there will be 2 files opened in the editor, but one is selected. I can add a comment to it.

File: src/com/twitter/intellij/pants/compiler/actions/PantsCompileCurrentTargetAction.java (hunk 2 above)
Comment (wisechengyi) [inline src/com/twitter/intellij/pants/compiler/actions/PantsCompileCurrentTargetAction.java]: Ah I see. I split the editor panes, then it will return targets associated with both files in editor. Let me see if there is a way to find the focused one.

File: src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java (hunk 4 above)
Comment (wisechengyi) [inline src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java]: It is related, because if user triggers the same action without any change, the IDE will noop, which makes a bit confusing.

Comment (wisechengyi) [inline src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java]: will change it to a notification since printing out in console may not be obvious either.

File: tests/com/twitter/intellij/pants/integration/OSSPantsCompileActionsTest.java (hunk 1 above)
Comment (wisechengyi) [inline tests/com/twitter/intellij/pants/integration/OSSPantsCompileActionsTest.java]: corrected.

File: src/com/twitter/intellij/pants/compiler/actions/PantsCompileCurrentTargetAction.java (hunk 2 above)
Comment (wisechengyi) [inline src/com/twitter/intellij/pants/compiler/actions/PantsCompileCurrentTargetAction.java]: corrected to the editor in focus.
